person.rb

```diff
@@ -1,14 +1,52 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class BaseDecorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    return @nameable.correct_name[0...10] if @nameable.correct_name.length > 10
+
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Instead of repeating your code, you can import the `name.rb`  file to this file, kindly consider removing this code and import it instead so your following best practices